Bug: SvelteKit build in node docker image fails - Cannot find module './$houdini'

### Describe the bug

Hello 👋 

When trying to build our SvelteKit application with a Dockerfile for deployment on the server, we currently receive an error when running the step `svelte-check`: "Error: Cannot find module './$houdini' or its corresponding type declarations. (ts)". Removing this step from the Dockerfile leads to an successful build, but then the application breaks during runtime: "Undefined property pageInfo" (using a pagination setup here).

After quite a lot of debugging, I figured out that the `houdini generate` command does not write all of the necessary files. Within the generated `$houdini` folder, the path `types/src/routes` is empty, where it should actually create all of the `.d.ts` files describing the available data for each page. On the other hand, the `$houdini/artifacts` folder is filled correctly. Why those files have not been written remains unknown. The standard SvelteKit type files (`.svelte-kit/types/src/routes/...`) are generated properly.

This issue only occurs when building the application in the environment of the node image. On my local system (macOS) there is no such issue. We make use of groups in the folder-based router (e.g. `src/routes/(protected)/articles/+page.svelte` and thought this might be an issue, but a route such as `src/routes/api/...` is not processed as well.

I hope my explanations above clarify the issue properly. If there is any other information required for debugging and analysis, please just let me know!

Any help in fixing this issue is very much appreciated 🙏  Thanks!

### Severity

blocking all usage of Houdini

### Steps to Reproduce the Bug

1. Setup houdini with SvelteKit as per the official documentation.
2. Create a Dockerfile to build the product using the official node docker image.
3. Run `svelte-check` as part of the process to detect this error early on (otherwise it would come up at runtime).
4. Tested with all recent versions of houdini and houdini-svelte (0.18.0 -> 1.0.0-next.1).


### Reproduction

_No response_

--- a/.changeset/silly-spiders-grin.md
+++ b/.changeset/silly-spiders-grin.md
@@ -1,0 +1,5 @@
+---
+'houdini-svelte': patch
+---
+
+Fix build issue with docker

--- a/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
+++ b/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
@@ -21,8 +21,6 @@ export default async function svelteKitGenerator(
 		return
 	}
 
-	console.log('generating')
-
 	await walk_routes(config, framework, {
 		async route({
 			dirpath,

--- a/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
+++ b/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
@@ -21,6 +21,8 @@ export default async function svelteKitGenerator(
 		return
 	}
 
+	console.log('generating')
+
 	await walk_routes(config, framework, {
 		async route({
 			dirpath,
@@ -30,9 +32,9 @@ export default async function svelteKitGenerator(
 			layoutExports,
 			pageExports,
 		}) {
-			// remove testing later
 			const relativePath = path.relative(config.routesDir, dirpath)
 			const target = path.join(type_route_dir(config), relativePath, config.typeRootFile)
+			console.log('step', target, svelteTypeFilePath)
 
 			const houdiniRelative = path.relative(target, config.typeRootDir)
 
@@ -246,6 +248,8 @@ export default async function svelteKitGenerator(
 			}
 		},
 	})
+
+	throw new Error('done')
 }
 
 function getTypeImports(

--- a/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
+++ b/packages/houdini-svelte/src/plugin/codegen/routes/index.ts
@@ -34,7 +34,6 @@ export default async function svelteKitGenerator(
 		}) {
 			const relativePath = path.relative(config.routesDir, dirpath)
 			const target = path.join(type_route_dir(config), relativePath, config.typeRootFile)
-			console.log('step', target, svelteTypeFilePath)
 
 			const houdiniRelative = path.relative(target, config.typeRootDir)
 
@@ -248,8 +247,6 @@ export default async function svelteKitGenerator(
 			}
 		},
 	})
-
-	throw new Error('done')
 }
 
 function getTypeImports(

--- a/packages/houdini-svelte/src/plugin/kit.ts
+++ b/packages/houdini-svelte/src/plugin/kit.ts
@@ -17,7 +17,6 @@ export function is_route(config: Config, framework: Framework, filepath: string)
 		return false
 	}
 
-	// only consider filepaths in src/routes
 	if (
 		!filepath.startsWith(config.routesDir) &&
 		!filepath.startsWith(path.join(config.projectRoot, config.routesDir))
@@ -119,7 +118,9 @@ export async function walk_routes(
 	config: Config,
 	framework: Framework,
 	visitor: RouteVisitor,
-	dirpath = config.routesDir
+	dirpath = config.routesDir.startsWith(config.projectRoot)
+		? config.routesDir
+		: path.join(config.projectRoot, config.routesDir)
 ) {
 	let pageExports: string[] = []
 	let layoutExports: string[] = []
@@ -267,11 +268,12 @@ export async function walk_routes(
 		const relative_path_regex = /src(.*)/
 
 		// here we define the location of the correspoding sveltekit type file
+		const local = dirpath.replace(config.projectRoot, '')
 		const svelteTypeFilePath = path.join(
 			config.projectRoot,
 			'.svelte-kit',
 			'types',
-			dirpath.match(relative_path_regex)?.[0] ?? '',
+			local.match(relative_path_regex)?.[0] ?? '',
 			'$types.d.ts'
 		)
 

--- a/packages/houdini-svelte/src/plugin/kit.ts
+++ b/packages/houdini-svelte/src/plugin/kit.ts
@@ -18,7 +18,10 @@ export function is_route(config: Config, framework: Framework, filepath: string)
 	}
 
 	// only consider filepaths in src/routes
-	if (!filepath.startsWith(config.routesDir)) {
+	if (
+		!filepath.startsWith(config.routesDir) &&
+		!filepath.startsWith(path.join(config.projectRoot, config.routesDir))
+	) {
 		return false
 	}
 

--- a/packages/houdini-svelte/src/plugin/kit.ts
+++ b/packages/houdini-svelte/src/plugin/kit.ts
@@ -118,9 +118,7 @@ export async function walk_routes(
 	config: Config,
 	framework: Framework,
 	visitor: RouteVisitor,
-	dirpath = config.routesDir.startsWith(config.projectRoot)
-		? config.routesDir
-		: path.join(config.projectRoot, config.routesDir)
+	dirpath = config.routesDir
 ) {
 	let pageExports: string[] = []
 	let layoutExports: string[] = []

